From 4de457f68638bead4a5bf4098bb7efc2c82bb579 Mon Sep 17 00:00:00 2001
From: Michal Marek <mmarek@suse.cz>
Date: Thu, 5 Nov 2015 13:42:52 +0100
Subject: [PATCH] kabi: Hide change to struct loop_device
Patch-mainline: Never, kabi
References: bsc#931634, fate#319010

The structure is declared in drivers/block/ for use by cryptoloop, so
there is no clean way for external modules to use it. Move it at the end
of the struct and hide it from the checker.

Signed-off-by: Michal Marek <mmarek@suse.cz>
---
 drivers/block/loop.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/block/loop.h b/drivers/block/loop.h
index fefe3dce3959..ba126f1670f2 100644
--- a/drivers/block/loop.h
+++ b/drivers/block/loop.h
@@ -47,7 +47,6 @@ struct loop_device {
 	struct file *	lo_backing_file;
 	struct block_device *lo_device;
 	unsigned	lo_blocksize;
-	unsigned	lo_logical_blocksize;
 	void		*key_data; 
 
 	gfp_t		old_gfp_mask;
@@ -64,6 +63,9 @@ struct loop_device {
 
 	struct request_queue	*lo_queue;
 	struct gendisk		*lo_disk;
+#ifndef __GENKSYMS__
+	unsigned	lo_logical_blocksize;
+#endif
 };
 
 /* Support for loadable transfer modules */
-- 
2.1.4

